File: application/app/src/main/java/com/yahyeet/boardbook/activity/IFutureInteractable.java
package com.yahyeet.boardbook.activity;

/**
 * A view element that can interact with CompletableFutures
 */
public interface IFutureInteractable {

	/**
	 * Enables all interactive elements of the activity
	 */
	void enableViewInteraction();

	/**
	 * Disables all interactive elements of the activity
	 */
	void disableViewInteraction();

	/**
	 * Displays error message of games not loading correctly
	 */
	void displayLoadingFailed();
}

File: application/app/src/main/java/com/yahyeet/boardbook/activity/IFutureInteractable.kt
package com.yahyeet.boardbook.activity

interface IFutureInteractable {

    /**
     * Enables all interactive elements of the activity
     */
    fun enableViewInteraction()

    /**
     * Disables all interactive elements of the activity
     */
    fun disableViewInteraction()

    /**
     * Displays error message of games not loading correctly
     */
    fun displayLoadingFailed()
}
